src/main/java/reactor/kafka/receiver/internals/DefaultKafkaReceiver.java

```diff
@@ -153,30 +156,37 @@ public DefaultKafkaReceiver(ConsumerFactory consumerFactory, ReceiverOptions<K,
         return flux
                 .map(consumerRecords -> Flux.fromIterable(consumerRecords)
                                             .doAfterTerminate(() -> {
-                                                    for (ConsumerRecord<K, V> r : consumerRecords)
-                                                        new CommittableOffset(r).acknowledge();
-                                                }));
+                                                for (ConsumerRecord<K, V> r : consumerRecords)
+                                                    new CommittableOffset(r).acknowledge();
+                                            }));
     }
 
     @Override
     public Flux<ConsumerRecord<K, V>> receiveAtmostOnce() {
         this.ackMode = AckMode.ATMOST_ONCE;
         atmostOnceOffsets = new AtmostOnceOffsets();
-        Flux<ConsumerRecord<K, V>> flux = createConsumerFlux()
-                .concatMap(consumerRecords -> Flux.fromIterable(consumerRecords));
-        return withDoOnRequest(flux)
-                .doOnNext(r -> {
+        return createConsumerFlux()
+            .concatMap(cr -> Flux
+                    .fromIterable(cr)
+                    .concatMap(r -> {
```

Comment:
`.delayUntil` here would help to get rid of `return Mono.just(r)` and `.then(Mono.just(r))` inside the method

Reply:
done